sycl/include/CL/sycl/INTEL/esimd/detail/esimd_memory_intrin.hpp

```diff
@@ -466,7 +466,7 @@ __esimd_raw_send_store(uint8_t modifier, uint8_t execSize,
 
 template <typename Ty, int N, int NumBlk, sycl::INTEL::gpu::CacheHint L1H,
           sycl::INTEL::gpu::CacheHint L3H>
-SYCL_EXTERNAL sycl::INTEL::gpu::vector_type_t<
+inline sycl::INTEL::gpu::vector_type_t<
```

Comment:
With this change, for non-sycl-device build, compiler FE will see one SYCL_EXTERNAL declaration and one inline definition for the same API, while the existing code made it consistent. What is the exact FE rule here, or compiler doesn't have diagnostic for this? Should similar change be made in other files such as esimd_math_intrin.hpp?

Reply:
`inline` will be seen only by the host compiler, which is really the intent.
Device compiler will continue to see SYCL_EXTERNAL w/o definition

Reply:
> Should similar change be made in other files such as esimd_math_intrin.hpp?

Yes. This is achieved by including `inline` into `ESIMD_INLINE` used in other files.
Also, SYCL_EXTERNAL is defined to empty string for the host compiler, so `void foo` followed by `inline void foo` should be OK.

Reply:
Had a sync with @kychendev - esimd_math_intrin.hpp should be really fixed, my omission. Will do.